### PR TITLE
Add auto labeler for pull requests

### PR DESCRIPTION
**Pull request by edg-l**
This PR adds a github action that automatically labels PRs based on files changed: https://github.com/actions/labeler

---

Someone in the ddnet team needs to add an actions secret with the name GITHUB_TOKEN to https://github.com/ddnet/ddnet/settings/secrets/actions , can generate one from: https://github.com/settings/tokens

I would do it, but i think @def- is the best fit for such things.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- any: ["src/macos/**/*", "./**/*.mm"]
+
+tests:
+- any: ["src/test/**/*"]
+
+tools:
+- any: ["src/tools/**/*]
+
+"UI/UX":
+- any: ["src/game/client/components/menu_*.{h,cpp}", "src/game/client/components/*board.{h,cpp}", "src/game/client/components/{hud,chat,freezebars,killmessages}.{h,cpp}"]
```
Comment (Robyt3) [inline .github/labeler.yml:47]: Also include `src/game/client/ui*.{h,cpp}`

**Comment by heinrich5991** [comment]
What problem does this solve? In the PR template, you're encouraged to write a motivation for changes.

**Comment by edg-l** [comment]
> What problem does this solve? In the PR template, you're encouraged to write a motivation for changes.

people cant add labels to prs if they are not members, this does it for them, i think its p obvious

**Comment by heinrich5991** [comment]
It wasn't obvious to me, that's why I asked. Why do we want to label pull requests?

**Comment by def-** [comment]
We should quickly submit or request changes on all pull requests, not have them around for long enough that we require tags. Tags for issues make more sense, since we seem to accumulate them (unfortunately).

**Comment by edg-l** [comment]
they may be useful to search and filter the pull requests in the future, not everything is meant for the present, and adding these labels may help future devs search related prs, e.g to the server, client, networking, and they may find help in implementing future features.

**Comment by edg-l** [comment]
this adds 0 complexity to the pull request author and helps future searchers, i don't think its something bad to have labels vs not having them

**Comment by heinrich5991** [comment]
The maintenance of another tool doesn't seem worth it, to me. If we find a concrete use case, I'm open to listen to it.

**Comment by edg-l** [comment]
It seems worth to me, it's almost no maintenance and I listed good use cases.

Having labels > not having labels

but it seems you are keen on not having this even after listing the reasons.

> What problem does this solve? In the PR template, you're encouraged to write a motivation for changes.
Btw most prs i see start exactly like i have done, and i don't see you asking this

**Comment by heinrich5991** [comment]
> they may be useful to search and filter the pull requests in the future

This is too vague for me. I don't understand why these labels help filter pull requests. Can you give a concrete example where these labels would be helpful?

> and adding these labels may help future devs search related prs, e.g to the server, client, networking, and they may find help in implementing future features.

This would also be helped by an example. When I search for related PRs, I usually check the commit history of a file, or look at the blame view.

> Btw most prs i see start exactly like i have done, and i don't see you asking this

Feel free to ask for a motivation in this case. I probably haven't because I could infer the motivation.

**Comment by edg-l** [comment]
> > they may be useful to search and filter the pull requests in the future
> 
> This is too vague for me. I don't understand why these labels help filter pull requests. Can you give a concrete example where these labels would be helpful?

If I want to search for a pull request that touches server-side code i could simply do so with the label, which probably may lead to better results than a text query because they don't contain the keyword "server" on it.

> > and adding these labels may help future devs search related prs, e.g to the server, client, networking, and they may find help in implementing future features.
> 
> This would also be helped by an example. When I search for related PRs, I usually check the commit history of a file, or look at the blame view.

A new developer may not know directly the file before implementing something, hence why a label for PRs that are for example "server" may help them find server-related features and gather ideas from there to see how to implement stuff. This doesn't require prior knowledge on which file stuff is located.

**Comment by heinrich5991** [comment]
> If I want to search for a pull request that touches server-side code i could simply do so with the label, which probably may lead to better results than a text query because they don't contain the keyword "server" on it.

Why is it interesting to look for pull requests that contain server-side code? To take it ad absurdum, I could create a tag that is applied to every PR that changes a line with the word "tee" in it.

If I want to search for a pull request that changes something with the word tee, I could simply do so with the label, etc.

To me, these labels seem like a solution looking for a problem.

> A new developer may not know directly the file before implementing something, hence why a label for PRs that are for example "server" may help them find server-related features and gather ideas from there to see how to implement stuff. This doesn't require prior knowledge on which file stuff is located.

Sounds good, we could use some documentation on that. I don't think "labels on PRs" is the correct place for this documentation.

**Pull request closed without merging**